### PR TITLE
fix(#3844): Add service discovery for Kamelet data type converter

### LINKED ISSUE #3844: Add data type converter factory finder discovery in camel-quarkus
Kamelets introduces a data type concept as described in #1980

The data type implementations are resolved with factory finder mechanism. We will need to add the factory finder for DataTypeConverters when building the camel-quarkus jar for the integration. This makes sure that META-INF resources for factory finder lookup are included in the jars and runtime is able to resolve the data type implementation at runtime.

Usually this can be done with adding `quarkus.camel.service.discovery.include-patterns` property at build time.

See https://github.com/apache/camel-kamelets/pull/1162 for details

### PR DESCRIPTION
**Pull request by christophd**
- Enable service discovery for data type converter resources in order to enable factory finder mechanism when resolving data type implementations
- Add proper quarkus-maven-plugin build time properties for quarkus.camel.* properties
- Fixes the way camel-quarkus build time properties are set (set properties on the quarkus-maven-plugin instead of using generic Maven system properties)
- Explicitly add quarkus.camel.service.discovery.include-patterns for data type converter resources in order to enable lazy loading of Kamelets data type implementations

Relates to [#1980](https://github.com/apache/camel-k/issues/1980)

also see https://github.com/apache/camel-kamelets/tree/main/library/camel-kamelets-utils/src/main/resources/META-INF/services/org/apache/camel/datatype/converter

Fixes #3844

### CODE REVIEW

**Review by squakez — COMMENTED** [review]
I miss an E2E test to validate the feature you're trying to enable.

**Comment by christophd** [comment]
@squakez I have added some e2e YAKS test with Kamelets using the service discovery mechanism

the failing `install-it` job seems to be non related to this PR

File: pkg/apis/camel/v1/maven_types.go [hunk 1]
```diff
@@ -96,4 +96,9 @@ type Server struct {
 	Configuration Properties `xml:"configuration,omitempty" json:"configuration,omitempty"`
 }
 
-type Properties map[string]string
```
Comment (squakez) [inline pkg/apis/camel/v1/maven_types.go:99]: This change here looks to be breaking compatibility in the API. In order to be performed we should either have a backward compatible field or deprecate the previous one and make it coexist for a couple of versions with the newer.

**Review by squakez — CHANGES_REQUESTED** [review]

File: pkg/apis/camel/v1/maven_types.go (hunk 1 above)
Comment (christophd) [reply pkg/apis/camel/v1/maven_types.go:99]: @squakez I refactored the changes to only use the new plugin properties structure in Maven plugin execution configuration. The other occurrences of properties (e.g. in server configuration) now remain unchanged.

**Comment by christophd** [comment]
@squakez please have another look

File: pkg/apis/camel/v1/maven_types.go (hunk 1 above)
Comment (squakez) [reply pkg/apis/camel/v1/maven_types.go:99]: I wonder if we could reach the same level of compatibility having `type Properties map[string]interface{}`. In this way, we don't need to include a new type and we neither have to deprecate the old one.

Comment (christophd) [reply pkg/apis/camel/v1/maven_types.go:99]: Yeah, I have tried this one, too. It fails in client generator as it is not allowed to use generic interface{} at this point

**Review by squakez — APPROVED** [review]

**Pull request merged**